sdk/core/azure-core/inc/azure/core/http/websockets/websockets_transport.hpp

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Utilities to be used by HTTP WebSocket transport implementations.
+ */
+
+#pragma once
+
+#include "azure/core/context.hpp"
+#include "azure/core/http/http.hpp"
+
+namespace Azure { namespace Core { namespace Http { namespace WebSockets {
+
+  /**
+   * @brief Base class for all WebSocket transport implementations.
+   */
+  class WebSocketTransport : public HttpTransport {
+  public:
+    /**
+     * @brief Web Socket Frame type, one of Text or Binary.
+     */
+    enum class WebSocketFrameType
+    {
+      /**
+       * @brief Indicates that the frame is a partial UTF-8 encoded text frame - it is NOT the
+       * complete frame to be sent to the remote node.
+       */
+      FrameTypeTextFragment,
+      /**
+       * @brief Indicates that the frame is either the complete UTF-8 encoded text frame to be sent
+       * to the remote node or the final frame of a multipart message.
+       */
+      FrameTypeText,
+      /**
+       * @brief Indicates that the frame is either the complete binary frame to be sent
+       * to the remote node or the final frame of a multipart message.
+       */
+      FrameTypeBinary,
+      /**
+       * @brief Indicates that the frame is a partial binary frame - it is NOT the
+       * complete frame to be sent to the remote node.
+       */
+      FrameTypeBinaryFragment,
+
+      FrameTypeClosed,
+    };
+    /**
+     * @brief Destructs `%HttpTransport`.
+     *
+     */
+    virtual ~WebSocketTransport() {}
+
+    /**
+     * @brief Determines if the transport natively supports WebSockets or not.
+     *
+     * @returns true iff the transport has native websocket support, false otherwise.
```

Comment:
```suggestion
     * @returns true if the transport has native websocket support, false otherwise.
```

Reply:
This isn't a typo - "iff" is shorthand for "if and only if".